### PR TITLE
Refactor compute parents post state

### PR DESCRIPTION
**Pull request by KentShikama**
Fix the fact we call indexOf on an Option[Int] instead of
the actual Int.

Group ordering with the calls that require implicit ordering.

Clean up some comments.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala [hunk 1]
```diff
@@ -173,54 +173,68 @@ object InterpreterUtil {
...
+      initStateHash: StateHash
+  )(implicit scheduler: Scheduler): F[Either[Throwable, StateHash]] =
+    for {
+      parentsMetadata <- parents.toList.traverse(b => dag.lookup(b.blockHash).map(_.get))
+      ordering        <- dag.deriveOrdering(0L) // TODO: Replace with an actual starting number
+      blockHashesToApply <- {
+        implicit val o: Ordering[BlockMetadata] = ordering
+        for {
+          uncommonAncestors          <- DagOperations.uncommonAncestors[F](parentsMetadata.toVector, dag)
+          ancestorsOfInitParentIndex = 0
```
Comment (lukasz-golebiewski) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala]: This doesn't change between iterations so can be moved to a constant at the beginning of the method

Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala]: There is no iterations going on here.

**Comment by ArturGajowy** [comment]
hmmm... that's unfortunate. I guess the way to stop bors would be `bors r-`. I'm also not sure if bors didn't check the review statuses again, or was just happy with there being at least one :heavy_check_mark:. Should we revert that commit, or just have a follow-up PR fixing @rabbitonweb's comment?

File: casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala [hunk 2]
```diff
@@ -173,54 +173,68 @@ object InterpreterUtil {
...
+            .toVector
+            .sorted // Ensure blocks to apply is topologically sorted to maintain any causal dependencies
+        } yield result
+      }
+      blocksToApply <- blockHashesToApply.traverse(b => ProtoUtil.unsafeGetBlock[F](b.blockHash))
+      deploys       = blocksToApply.flatMap(_.getBody.deploys.flatMap(ProcessedDeployUtil.toInternal))
+    } yield
+      runtimeManager
+        .replayComputeState(initStateHash, deploys, time)
+        .runSyncUnsafe(Duration.Inf) match {
```
[the remark this answers is not in the document]
Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala]: @odeac RHOL-1050 will fix this. We should really let him work on merging it.

**Comment by KentShikama** [comment]
In any case don't revert this PR, we need the changes in here. The runSyncUnsafe is nothing new.

**Comment by KellyatPyrofex** [comment]
This PR supports work in [RHOL-1119](https://rchain.atlassian.net/browse/RHOL-1119)

**Pull request merged**